DEV_GUIDELINES.md

```diff
@@ -106,6 +132,16 @@ Views represent a page or a section of a page in IDSeq. Views are stored in the
 When designing your view, try to make as much use of the components in the `ui `and `visualizations` folders. This should be mostly plug-and-play components. 
 If you need to place a ui component in a customized place for your view (i.e. if the layout elements cannot help you), wrap it in a `div` with a proper class name, and style it in the view's css file.
 
+**Other Best Practices**
+
+* Always create a new object when modifying component state. This allows React to figure out if props have changed by via shallow comparison, and allows for future rendering optimizations using `React.PureComponent`. `lodash/fp` functions are immutable and do this by default.
```

Comment:
Maybe emphasize the key sentence on these bullets like `**create a new object when modifying component state**`